dev/src/util.rs

```diff
@@ -155,18 +155,20 @@ pub async fn invalid_image_spawn_request() -> SpawnRequest {
 pub fn base_scheduler_request() -> ScheduleRequest {
     ScheduleRequest {
         cluster: ClusterName::new("plane.test"),
-        metadata: vec![("foo".into(), "bar".into())].into_iter().collect(),
-        backend_id: None,
-        max_idle_secs: Duration::from_secs(10),
-        executable: DockerExecutableConfig {
-            env: vec![("PORT".into(), "8080".into())].into_iter().collect(),
-            image: TEST_IMAGE.into(),
-            credentials: None,
-            resource_limits: Default::default(),
-            pull_policy: Default::default(),
-            port: None,
-            volume_mounts: vec![],
-        },
-        require_bearer_token: false,
+        resource: Resource::Backend(BackendResource {
```

Comment:
This breaks the existing drone API, right? i.e. a running drone would not work if we updated the controller version?

Reply:
Yep, I'm going to figure out how to get serde to let me keep the old way of doing things or have ScheduleResourceRequest be separate from ScheduleRequest.

But I want to know if this pr after fixes should be merged, or if I should try and coalesce things a bit such that instead of  having three different types there's traits on one type or something.